### ISSUE TITLE
style guide for lessons?

### ISSUE THREAD

**Issue by bbolker** [swcarpentry/r-novice-gapminder#251]
When I Google `"software carpentry" "style guide"` I can find references to Python and R coding style guides, but nothing about a style guide for the lessons themselves comes up.  I'm trying to decide about picky things (after all, that's what a style guide is about ...) like:

- when to use code formatting (back-ticks): functions, variables, packages ...
- should functions be `foo()` or `foo` ?
- 'csv file' or 'CSV file' ?
- explicit `return()` statements  (this is a coding style question, but to my surprise this isn't explicitly stated in [Google's R style guide](https://google.github.io/styleguide/Rguide.xml) 
 ... etc.

Is there a central resource for this, or interest in creating one?

**Comment by naupaka** [comment]
Hi @bbolker !

The closest thing we have is this for the lessons: http://swcarpentry.github.io/lesson-example/

Seems like an episode in that lesson would probably be the best place to stick a SWC style guide.

For the most part, for the R lessons we try to follow the [Google R style guide](https://google.github.io/styleguide/Rguide.xml) as you mentioned or [Hadley's style guide](http://adv-r.had.co.nz/Style.html) from Advanced R.

Generally I have been asking people for code-style formatting for functions and variables, but haven't been too consistent with package names. Usually yes, though.

Generally I always request `foo()` instead of `foo`, following how it's done in the Google style guide.

No official position on CSV vs csv AFAIK, but I prefer csv myself.

Generally haven't been requesting explicit `return()` statements.

In conclusion, seems like there indeed would be a use and need for codifying some of this. If there is something more formal than what I mentioned, I'm not aware of it.

**Comment by jcoliver** [comment]
Closing this for now, but please feel free to submit a pull request if time and resources afford.
